### DISCUSSION TITLE
Is it possible to export DIPY visualize result to .trk with scalar

### DISCUSSION THREAD

**Discussion by iPsych** [dipy/dipy#2660] [Scientific Visualization]
Hello,
DIPY seems as the only solution to 1D array (i.e. p-value) to .trk tract bundle as scalar.
Since usually the streamlines of .trk have different length, applying 1D array to tract bundle is far tricky compare to single 'mean tract' tube. 
https://dipy.org/documentation/1.5.0/examples_built/bundle_assignment_maps/

BUAN example of DIPY was the only proper solution I've found (any other recommended way?).
One problem is that the result only saved as .png.
Is it possible to the DIPY visualization result to .trk with scalar data?
(In above example, actor.line with disc_color to .trk)

**Comment by Garyfallidis** [comment]
Hi @iPsych yes of course it is possible. But you have to help us a bit here reducing our response time by posting your existing code and we can provide updates. Is that okay?

**Reply by iPsych** [reply]
> Hi @iPsych yes of course it is possible. But you have to help us a bit here reducing our response time by posting your existing code and we can provide updates. Is that okay?

Hi @Garyfallidis,
The goal is save .trk including save_tractogram and stateful_tractogram with scalar value (color) like the visualization in the example link. The reason to use the dipy, is the 'bundle_assignment_maps' seems have 'aligned' scalar position no matter to the various length of each streamlines.
https://dipy.org/documentation/1.5.0/examples_built/bundle_assignment_maps/

**Comment by iPsych** [comment]
@Garyfallidis 
Below I attached the test.
Two figures are exactly same p-value array overlaid on the .trk streamline.
The first one is not properly represented when just merged to .trk file using along-the-tract Matlab approach. (In along-the-tract, usually overlay the 1 dimensional scalar array to the 'mean' tract, which consists in a single streamline.)  
However, dipy solution (2nd picture) works perfectly.

Just made disks_color2 variable from scalar array and run BUAN example as below.
```
scene.add(actor.line(model_af_l, fake_tube=True, colors=disks_color2,
                     linewidth=6))
scene.set_camera(focal_point=(-18.17281532, -19.55606842, 6.92485857),
                 position=(-360.11, -30.46, -40.44),
                 view_up=(-0.03, 0.028, 0.89))
window.record(scene, out_path='rh_ccg_after_assignment_maps.png',
              size=(600, 600))
if interactive:
    window.show(scene)
```

One single issue is there is no way to export the .trk with the color scalar as displayed.
I dig a bit, but the dipy rely on fury to visualize, and fury rely on vtk visualization engine.
So, it's very confusing where the 'center-fitting' happens.

If it's possible to pass the actor.line to StatefulTractogram, than the io module of dipy will works.

<img width="578" alt="Screen Shot 2022-10-29 at 2 41 23 AM" src="https://user-images.githubusercontent.com/28276515/198699307-2238e645-e053-4b8c-bc80-10320eca9ff4.png">

![rh_ccg_after_assignment_maps](https://user-images.githubusercontent.com/28276515/198699361-cd8467a5-69ac-4af1-a5b6-e8487e0a68a5.png)
